app/src/main/java/org/ea/sqrl/activites/LoginActivityHelper.java

```diff
@@ -0,0 +1,73 @@
+package org.ea.sqrl.activites;
+
+import android.hardware.biometrics.BiometricPrompt;
+import android.os.Build;
+import android.os.CancellationSignal;
+
+import org.ea.sqrl.R;
+import org.ea.sqrl.activites.base.LoginBaseActivity;
+import org.ea.sqrl.processors.BioAuthenticationCallback;
+import org.ea.sqrl.processors.CommunicationFlowHandler;
+import org.ea.sqrl.processors.SQRLStorage;
+
+import java.security.KeyStore;
+
+import javax.crypto.Cipher;
+
+public class LoginActivityHelper {
```

Comment:
Why move a function used ones to a separate class when this case doesn't have a required version for running?

And why check for the version both at the call and the definition of the function?

Reply:
Good questions, both of them.  I struggled with both of those questions when I was working on it.

The separate class is just a weird thing about running the application.  There is no difference in the code logic, but if that bio-metric code is in a class that does not extend `Activity`, then the stack traces do not appear.  It's a work-around for what I consider a Google bug.

Checking the version in both places: Leaving the version check in the original place was for clarity.  Checking the version in the new class was in case the method was called from somewhere else.  Could be argued that it's confusing, or wasteful.  But I was going for safety.